### DISCUSSION TITLE
How to preserve a file name when saving a page in the browser?

### DISCUSSION THREAD

**Discussion by jacmgr** [datenstrom/community#962] [Ask a question] [answered]
Using the incredible EDIT extension.  For example when I edit an existing file with an existing name such as "README-Todo.md".  If I make changes and then SAVE the editor has renamed it "20240323-to-do-tasks.md".  I see how new name is based off of meta data in the file.  My problem is that I have many internal links that are fixed to a certain filename such as "README-Todo.md" but that file no longer exists.

How can I set the behavior so that when Edit extension save an existing file, the filename is preserved?

Thankyou.
john

**Comment by pftnhr** [comment]
According to the [page settings](https://datenstrom.se/yellow/help/how-to-change-the-system#page-settings) you could use `TitleSlug`.

**Comment by jacmgr** [comment]
Thankx for that idea. Unfortunately that just does not make sense to me.   Markdown is intended to be portable.  So, I use "TitleSlug: README" per the documentation. Then when i edit my README.md in yellow, it changes the name on the disk to "2024-03-23-readme.md"  Internally to Yellow, it might still find the proper location "README", but when I upload the files to a repo or share them some other way, the "README.md" won't exist anymore in the group of files.

Similarly if I share the markdown files with someone who has a different simple markdown file browser; the internal links won't work anywhere except inside yellow.    I'll keep thinking on this and see if i can find a work around or other documentations.

**Reply by annaesvensson** [reply]
Thanks for the feedback. Apparently the editor is doing something that doesn't make sense in your workflow. Actually, the editor should check whether meta data has been changed before changing the file name. If you only edit the content of a page, the file name should not change when saving the file. It would be nice to have more information. Can you add an example Markdown file, including meta data, to your question above? What happens if you don't change the meta data?

Tip: You can add an example, screenshot and further information to your question.

**Reply by GiovanniSalmeri** [reply]
Just two sidenotes:

1. I do not think that portability of markdown files is among the design principles of Yellow. For example, all shortcuts obviously are not meant to work outside Yellow (I myself wrote and use always an [extension which replaces the normal syntax of links](https://github.com/GiovanniSalmeri/yellow-link)). In general, I guess that any CMS that uses markdown as input format has its own peculiarities.
2. Markdown, even if now used for many purposes, was originally meant as a simple way for writing text **which will be converted in HTML**. From this point of view, links with relative paths to .md files seem to me to be in a grey area. It is clear that `[beautiful file](https://example.org/file.pdf)` will become `<a href="https://example.org/file.pdf">beautiful file</a>`. But what should `[beautiful page](page.md)` become? `<a href="page.md">beautiful page</a>` or `<a href="page.html">beautiful page</a>` or something else? It is interesting that the [original description by John Gruber](https://daringfireball.net/projects/markdown/syntax#link) avoids any such example!

Probably all of this makes no difference for the problem discussed here and its possible solutions. Just I do not wonder if markdown files created or edited inside a CMS do not work fully outside it.

**Comment by jacmgr** [comment]
Interesting... Yes it seems to be occurring if i change/add "Published". If I change content it does NOT change filename. If I change or add other Meta Data Fields it does not change FileName.  So if that is the intended operation, the changing filenames won't work for my work flow/backup scenarios.  No worries, my easy solution is to not use the Meta Key "Published:", I'll just globally change it to "Released:" (or something like that) in my markdown files and my disk filenames should be fine.  That seems pretty good solution, compared to stop using the awesome EDIT extension. 

I am really enjoying using yellow the past 2 weeks.  Thankyou.

Interesting revelation of the behavior for page.md files.
   Page Location: /stories/
   Page FileName: content/10-stories/page.md
After Adding/Changing Published Meta
   Physical Folder Name: 2024-03-23-stories
   Page Location: /stories/
   Page FileName: content/2024-03-23-stories/page.md

**Reply by annaesvensson** [reply, marked as answer]
Thanks for the additional information. I am glad to hear you found a workaround. To be honest the numeric prefix for file names was intended for blog pages, it is triggered by the `Published` page setting, which we didn't expect people would use for something else than blog pages. I think the edit extension should distinguish whether a file name needs a numeric prefix or not.  Let us check how much source code it would take to implement, I'll get back to you.

**Comment by annaesvensson** [comment]
The [edit extension](https://github.com/annaesvensson/yellow-edit) has been updated and preserves the file name as you suggested. You can create file names with numeric prefix, for example the blog extension creates file names with the publication date, but this behaviour no longer happens everywhere. The edit extension can finally distinguish whether a file name needs a numeric prefix or not.

**Reply by jacmgr** [reply]
Wow. Very cool.  I'll check it out when I get back home in a couple days.
